clippy_lints/src/matches/redundant_guards.rs

```diff
@@ -80,6 +81,37 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, arms: &'tcx [Arm<'tcx>]) {
     }
 }
 
+fn is_the_local_binded_by_value_in_pat(local: &Expr<'_>, arm: &Arm<'_>) -> bool {
+    let local_id = if let ExprKind::Path(qpath) = local.kind
+        && let QPath::Resolved(_, Path {res, ..}) = qpath
+        && let Res::Local(id) = res
+    {
+        id
+    } else {
+        return false;
+    };
+
+    let Arm {
+        pat: Pat {
+            kind: PatKind::TupleStruct(_, pats, _),
+            ..
+        },
+        ..
+    } = arm
+    else {
+        return false;
+    };
+
+    for pat in *pats {
+        if let PatKind::Binding(annot, pat_id, ..) = &pat.kind
+            && matches!(annot.0, rustc_ast::ByRef::No)
```

Comment:
Would it make sense to have this ByRef check in the `get_pat_binding` function, instead of duplicating the code for finding the right binding in a new function? I.e. having it here:
https://github.com/rust-lang/rust-clippy/blob/415ba21c3b78125cf583ca0ef7a318a8f347a282/clippy_lints/src/matches/redundant_guards.rs#L89-L90